builder/oracle/classic/artifact.go

```diff
@@ -0,0 +1,46 @@
+package classic
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-oracle-terraform/compute"
+)
+
+// Artifact is an artifact implementation that contains a Snapshot.
+type Artifact struct {
+	Snapshot *compute.Snapshot
+	driver   *compute.ComputeClient
+}
+
+// BuilderId uniquely identifies the builder.
+func (a *Artifact) BuilderId() string {
+	return BuilderId
+}
+
+// Files lists the files associated with an artifact. We don't have any files
+// as the custom image is stored server side.
+func (a *Artifact) Files() []string {
+	return nil
+}
+
+func (a *Artifact) Id() string {
+	return a.Snapshot.Name
+}
+
+func (a *Artifact) String() string {
+	return fmt.Sprintf("A Snapshot was created: \n"+
```

Comment:
Useful attributes to return would be:
- ImageList `name` - e.g. `/Compute-acme/user@example.com/Packer_builder_test_12345678`
- ImageList Entry `version` e.g. 1
- MachineImage `file` -  e.g. `image/8b968684-6c7a-460c-a89e-c278851a0311-8e3a0496-d35c-43b9-874c-ff96b5db4093-snapshot.tar.gz`

Reply:
so I had a question about the ImageList entry versions -- should this be a user-required variable? Or should Packer check what is in the imagelist already and increment on top of that?

Reply:
I think packer should determine the next entry to use automagically. Might be nice to provide an optional attribute to set it, but I'm not sure ()haven't personally tested) what happens if you try to reuse a version, or try to create an entry out of sequence, so picking next available is probably less error prone.

Reply:
Yeah I checked it and the API gets unhappy if you try to set an already-used version; I've set it up to automagically increment.